controller/engine_image_controller.go

```diff
@@ -283,7 +284,19 @@ func (ic *EngineImageController) syncEngineImage(key string) (err error) {
 	if err != nil {
 		return errors.Wrapf(err, "cannot get daemonset for engine image %v", engineImage.Name)
 	}
+
+	isReady := true
```

Comment:
Still not sure why you need this boolean. Should this be covered by the reconciliation process?

Reply:
Or, maybe we should just put the creation/deletion in the `instance-manager-controller` instead, so it can be covered naturally.

Reply:
I want to make sure there is no expired instance manager before creating the related daemonset pod. 

If `syncInstanceManagers()` is executed, it means all daemonset pods are ready. Then it is meaningless to do cleanup.

Reply:
Yes. `isReady` means `engineImage.Status.State == types.EngineImageStateDeploying`. I can use this condition for simplicity.

We can add `isNodeDownOrDeleted()` to prevent overreacting for node down case. 

Deleting the instance manager pods will lead to marking the engine/replica of the instance manager as state `error`. It means cleanup can expose the error state immediately.

Reply:
1. Using `isNodeDownOrDeleted()` is not what I meant. The case should be covered by normal code path without introducing another condition?
2. I don't understand the third paragraph. And I meant `instance manager` (as regarding the engine image controller).

Reply:
Typically, the instance manager should be deleted if the related daemonset pod is not ready. But I don't want to delete the instance manager if the node is down. What's why I use `isNodeDownOrDeleted()` and the case cannot be covered by the normal code path.

For the third paragraph, I mean: the error state of engine/replica may not be exposed timely if we left instance manager pods there until we figured the engine daemon is ready or incompatible.

Reply:
If somehow one engine image pod is lost and it's rebuilding, do we need to stop the instance manager? Although the longhorn command won't work, and we cannot create new volumes on the node, the existing volumes may still be running?

Reply:
In this case, longhorn manager cannot communicate with the engine process via engine binary.  We cannot CRUD backup and snapshot besides creating volume.

Reply:
That's fine as long as user can still read/write the volume.

Reply:
OK. Then I can remove this cleanup